### PR TITLE
feat(issue-summary): Store on memcache

### PR DESCRIPTION
**Pull request by jennmueng**
Store the issue summary object in memcache instead of the group data

### CODE REVIEW

File: src/sentry/api/endpoints/group_ai_summary.py [hunk 1]
```diff
@@ -30,6 +32,7 @@ class SummarizeIssueResponse(BaseModel):
     group_id: str
     summary: str
     impact: str
+    headline: str
```
Comment (jennmueng) [inline src/sentry/api/endpoints/group_ai_summary.py:35]: includes adding new field returned from seer

File: tests/sentry/api/endpoints/test_group_ai_summary.py [hunk 2]
```diff
@@ -86,3 +108,8 @@ def test_ai_summary_call_seer(self, mock_get_event, mock_post):
             mock_response.json.return_value, snake_to_camel_case
         )
         mock_post.assert_called_once()
+        mock_cache.set.assert_called_once_with(
+            f"ai-group-summary:{group.id}",
+            mock_response.json.return_value,
+            timeout=60 * 60 * 24 * 7,  # 7 days
+        )
```
Comment (JoshFerge) [inline tests/sentry/api/endpoints/test_group_ai_summary.py:115]: wanna add one test where the cache isn't mocked?

File: src/sentry/api/endpoints/group_ai_summary.py [hunk 3]
```diff
@@ -109,9 +113,10 @@ def post(self, request: Request, group: Group) -> Response:
         if not features.has("organizations:ai-summary", group.organization, actor=request.user):
             return Response({"detail": "Feature flag not enabled"}, status=400)
 
-        if group.data.get("issue_summary"):
+        cache_key = "ai-group-summary:" + str(group.id)
+        if cache.has_key(cache_key):
```
Comment (wedamija) [inline src/sentry/api/endpoints/group_ai_summary.py:117]: You can just use `cache.get` once, like: 
https://github.com/getsentry/sentry/blob/605c24c35845a004926623d9c3c734c1046f62e2/src/sentry/http.py#L165-L166

File: tests/sentry/api/endpoints/test_group_ai_summary.py [hunk 4]
```diff
@@ -15,15 +15,18 @@ def _get_url(self, group_id: int):
         return f"/api/0/issues/{group_id}/summarize/"
 
     @patch("sentry.api.endpoints.group_ai_summary.GroupAiSummaryEndpoint._call_seer")
-    def test_ai_summary_get_endpoint_with_existing_summary(self, mock_call_seer):
+    @patch("sentry.api.endpoints.group_ai_summary.cache")
+    def test_ai_summary_get_endpoint_with_existing_summary(self, mock_cache, mock_call_seer):
```
Comment (wedamija) [inline tests/sentry/api/endpoints/test_group_ai_summary.py:19]: Personally I'd recommend not mocking the cache, but instead just store the value here using cache.set

**Review by wedamija — APPROVED** [review]

File: tests/sentry/api/endpoints/test_group_ai_summary.py (hunk 2 above)
Comment (jennmueng) [reply tests/sentry/api/endpoints/test_group_ai_summary.py:115]: done

File: tests/sentry/api/endpoints/test_group_ai_summary.py (hunk 4 above)
Comment (jennmueng) [reply tests/sentry/api/endpoints/test_group_ai_summary.py:19]: changed the test cases to all use the real cache

**Pull request merged**